File: android/app/src/main/kotlin/com/phobri/android/model/SmsMessage.kt
package com.phobri.android.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * SMS message synchronized with the desktop.
 */
@Serializable
data class SmsMessage(
    val id: Long = 0,
    val threadId: Long = 0,
    val address: String = "",
    val contactName: String? = null,
    val body: String = "",
    val date: Long = 0,
    val type: SmsType = SmsType.INBOX,
    val read: Boolean = true
) {
    /** Display-friendly name, falls back to address. */
    val displayName: String get() = contactName ?: address
}

@Serializable
enum class SmsType {
    @SerialName("inbox") INBOX,
    @SerialName("sent") SENT,
    @SerialName("draft") DRAFT,
    @SerialName("outbox") OUTBOX,
    @SerialName("failed") FAILED
}

/**
 * Call log entry synchronized with the desktop.
 */
@Serializable
data class CallLogEntry(
    val id: Long = 0,
    val number: String = "",
    val contactName: String? = null,
    val date: Long = 0,
    val duration: Long = 0,
    val type: CallType = CallType.INCOMING
) {
    /** Display-friendly name, falls back to number. */
    val displayName: String get() = contactName ?: number

    /** Duration formatted as m:ss. */
    val durationFormatted: String get() {
        val minutes = duration / 60
        val seconds = duration % 60
        return "$minutes:${seconds.toString().padStart(2, '0')}"
    }
}

@Serializable
enum class CallType {
    @SerialName("incoming") INCOMING,
    @SerialName("outgoing") OUTGOING,
    @SerialName("missed") MISSED,
    @SerialName("rejected") REJECTED,
    @SerialName("blocked") BLOCKED
}

File: desktop/Phobri.Desktop/Models/SmsMessage.cs
using System.Text.Json.Serialization;

namespace Phobri.Desktop.Models;

/// <summary>
/// Represents an SMS message synchronized from the phone.
/// </summary>
public sealed record SmsMessage
{
    [JsonPropertyName("id")]
    public long Id { get; init; }

    [JsonPropertyName("threadId")]
    public long ThreadId { get; init; }

    [JsonPropertyName("address")]
    public string Address { get; init; } = string.Empty;

    [JsonPropertyName("contactName")]
    public string? ContactName { get; init; }

    [JsonPropertyName("body")]
    public string Body { get; init; } = string.Empty;

    [JsonPropertyName("date")]
    public long Date { get; init; }

    [JsonPropertyName("type")]
    public SmsType Type { get; init; }

    [JsonPropertyName("read")]
    public bool Read { get; init; }

    /// <summary>
    /// Display-friendly contact name (falls back to address).
    /// </summary>
    [JsonIgnore]
    public string DisplayName => ContactName ?? Address;

    /// <summary>
    /// Date as DateTimeOffset for display.
    /// </summary>
    [JsonIgnore]
    public DateTimeOffset DateTime => DateTimeOffset.FromUnixTimeMilliseconds(Date);
}

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum SmsType
{
    Inbox,
    Sent,
    Draft,
    Outbox,
    Failed
}
